Description: Use clang14 for pybind11 instead of clang12
  See https://github.com/RobotLocomotion/ros-drake-vendor/issues/2
Applied-Upstream: https://github.com/RobotLocomotion/drake/pull/20732

--- a/tools/workspace/pybind11/libclang_setup.py
+++ b/tools/workspace/pybind11/libclang_setup.py
@@ -38,7 +38,7 @@ def add_library_paths(parameters=None):
                 parameters.append(sdkroot)
     elif platform.system() == 'Linux':
         arch = platform.machine()
-        library_file = f'/usr/lib/{arch}-linux-gnu/libclang-12.so'
+        library_file = f'/usr/lib/{arch}-linux-gnu/libclang-14.so'
     if not os.path.exists(library_file):
         raise RuntimeError(f'Library file {library_file} does NOT exist')
     cindex.Config.set_library_file(library_file)
